### PR TITLE
Clean up and expand autodiff in constructors.

### PR DESCRIPTION
**Pull request by pkofod**

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,6 +1,7 @@
 julia 0.5
 Calculus
 ForwardDiff 0.3.0 0.5.0
+ReverseDiff 0.0.3
```
Comment (anriseth) [inline REQUIRE]: Update this to v0.1.1?

Comment (pkofod) [inline REQUIRE]: Sure, just copied it over from your PR. Is the version limit actually needed?

Comment (anriseth) [inline REQUIRE]: I think v0.0.3 is needed, as v0.0.2 does not use `__precompile__()`

**Comment by pkofod** [comment]
Codecov... always here to ruin the party :( (better get testing...)

**Comment by pkofod** [comment]
@anriseth wonna have a last look?

File: src/objective_types.jl [hunk 2]
```diff
@@ -178,15 +207,15 @@ function TwiceDifferentiable{T}(f, g!, fg!, x_seed::Array{T}; autodiff = :finite
...
 end
 =#
-function TwiceDifferentiable(d::OnceDifferentiable; autodiff = :finitediff)
+function TwiceDifferentiable(d::OnceDifferentiable; autodiff = :finite)
     n_x = length(d.last_x_f)
     T = eltype(d.last_x_f)
-    if autodiff == :finitediff
+    if autodiff == :finite
         function h!(x::Vector, storage::Matrix)
             Calculus.finite_difference_hessian!(x->(d.f_calls[1]+=1;d.f(x)), x, storage)
```
Comment (anriseth) [inline src/objective_types.jl]: Should we remove the f_calls increment here?

**Comment by anriseth** [comment]
> @anriseth wonna have a last look?

Minus the f_calls increment in the finite difference hessian, this looks good to me :+1:

File: src/objective_types.jl (hunk 2 above)
Comment (pkofod) [inline src/objective_types.jl]: certainly!

**Pull request merged**